packages/components/src/range-control/index.native.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Internal dependencies
+ */
+import RangeCell from '../mobile/bottom-sheet/range-cell';
+
+function RangeControl( {
+	className,
+	currentInput,
+	label,
+	value,
+	instanceId,
+	onChange,
+	beforeIcon,
+	afterIcon,
+	help,
+	allowReset,
+	initialPosition,
+	min,
+	max,
+	...props
+} ) {
+	const id = `inspector-range-control-${ instanceId }`;
+
+	const currentInputValue = currentInput === null ? value : currentInput;
+
+	const initialSliderValue = isFinite( currentInputValue ) ? currentInputValue : initialPosition;
+
+	return (
+		<RangeCell
+			label={ label }
+			id={ id }
+			help={ help }
+			className={ className }
+			onChangeValue={ onChange }
+			aria-describedby={ !! help ? `${ id }__help` : undefined }
+			minimumValue={ min }
+			maximumValue={ max }
+			value={ value }
+			beforeIcon={ beforeIcon }
```

Comment:
It's ok when you are using separate `edit.js` and `edit.native.js`. Then you can pass different set of props and the problem is gone. I assume you are trying to use one common `edit.js` file both for web and mobile ? I wonder if it's the only prop that cause problems. I guess not ( I believe we are able to find more props e.g. `separatorType={ 'fullWidth' } ` can be passed to mobile but I'm afraid it's not desirable to obtain it on the web)

Reply:
> In the current draft PR for gallery, I'm using icon, but this won't work for web

What’s that icon for? Does it have the same purpose with either beforeIcon or afterIcon? @mkevins 

@jbinda right we have 1 edit.js. And we want to make components library cross platform, thus, we should be able to use same props for both platforms. Ofc there can be platform specific props but it would be necessary only when the actual behavior behimd that prop  is different. I was wondering, Is it possible to hide separatorType inside RangeControl native.js? If we’ll use the same separator type for every case I think we can do that.

Reply:
Btw, we have a [tech debt](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1421) to get rid of separatorType somehow.

Reply:
Hi @pinarol :wave: 

> What’s that icon for? Does it have the same purpose with either beforeIcon or afterIcon?

That `icon` prop is used by `Cell`, and it seems to be _similar_ to `beforeIcon`. `RangeControl` does not have this prop, though. I'm currently using it in the draft PR to prevent the slider layout from breaking (in the BottomSheet).

Reply:
@mkevins `icon` prop may look similar to `beforeIcon` bu in my opinion there is one important difference. Take a look [here](https://github.com/WordPress/gutenberg/pull/17282#issuecomment-535954687) and see examples from posted links there. Props `beforeIcon` and `afterIcon` is used to show what you are changing with slider (e.g. imagine how increase volume or screen brightness slider looks like). On the other hand `icon` is used to set up icon representation for BottomSheet Cell. I think we shouldn't mix that.

Another thing is that @lukewalczak already posted PR in which we split slider into 2 lines. In this version it's much clear to obtain the difference between what `icon` props stand for. I assume @mkevins you doesn't use this version of slider yet ?

@pinarol ok I'll try to take a look what else props could be problematic, check the tech debate and back to you with my thoughts

Reply:
Hi @jbinda :wave: 

Thanks for getting back to me.

> On the other hand icon is used to set up icon representation for BottomSheet Cell. I think we shouldn't mix that.

I agree :smile: 

> I assume @mkevins you doesn't use this version of slider yet ?

For the Gallery PR, my intent is to not use slider directly, but rather to use `RangeControl` as close to the web implementation as possible (since we'll be sharing `edit.js`). As for `icon`, I don't know that we'll be using an icon in the `BottomSheet` for the columns setting, but I've included it so far in the draft PR to keep the layout working for testing purposes.

Does Luke's PR supersede this one for `RangeControl`? If so, can you add a link here?

Reply:
> That icon prop is used by Cell, and it seems to be similar to beforeIcon. RangeControl does not have this prop, though. I'm currently using it in the draft PR to prevent the slider layout from breaking (in the BottomSheet). 

@mkevins That sounds a different kind of icon, could you send a screenshot? Should it be the same icon every time we use RangeControl? If so, we can hide it inside RangeControl.

Reply:
[Here](https://github.com/WordPress/gutenberg/pull/18157) is the styling PR which affects only the RangeCell component. Me and Luke discuss if my PR can broke something in his functionality and our conclusion was that the top layer in my PR (`RangeControl` component) just take the props, rename some of them if necessary to be consistent with usage on web and pass it down to RangeCell.

Because we also do the `{ ...props }` spreading we can pass anything we want and it can bring us to the situation you mentioned at the beginning. You will pass props specific for mobile to web as well. This rules also applied in way from web to mobile. As you already mentioned @pinarol possible solutions are:
1. Unify as many props as we can
2. Try to catch unwanted prop and spread in inside `index.native.js` and `index.js` file of `RangeControl`. It can be difficult because the current implementation allows to pass any prop you want.
3. Alternatively we can group props for mobile inside object like `mobileProps = { ... }` and spread it in mobile but separate it on web. Then all unified prop will pass to web and mobile and any custom prop that you add will go only to the web or custom prop wrapped in object go only to mobile.
Here is example what I mean:
![image](https://user-images.githubusercontent.com/21242757/67836240-637ef000-faec-11e9-8938-f14f4a2a25f1.png)

Reply:
Hmm actually we don't want to create `mobileProps`.  But we need to document mobile specific props. As far as I see currently we have the following, is there any other?
1. icon
2. seperatorType

We need to document them in the /range-control/README.md and state `Platform: Mobile`. 

And we need to make sure we are not breaking web when we pass these props, we should make necessary updates in index.js to prevent any crash. Same for web specific props, mobile app shouldn't be crashing even if they are passed.

Reply:
> @mkevins That sounds a different kind of icon, could you send a screenshot? Should it be the same icon every time we use RangeControl?

I've been arbitrarily using "admin-settings" icon while testing `BottomSheet.RangeCell`, and only have been using it in the Gallery draft with `RangeControl` a workaround for now. I'm not sure that we'll want to use an icon at all, or if we'd even need the extra prop to be exposed for mobile (e.g. there's no mention of it in the RangeCell design discussion, for example: https://github.com/WordPress/gutenberg/pull/18157). Here are some screenshots of how it currently looks with and without `icon` prop passed through to `RangeCell` for clarity:

With `icon="admin-settings"` | Without `icon` prop
---|---
![range-control-with-icon](https://user-images.githubusercontent.com/8507675/67916055-43305d80-fbe0-11e9-832a-2d173da755e2.png)|![range-control-without-icon](https://user-images.githubusercontent.com/8507675/67916315-04e76e00-fbe1-11e9-9dac-51b46bd8de12.png)

Reply:
@mkevins 
Hmmm I still wonder if you are using the most recent version of slider because after recent changes it should be more like [that[(https://github.com/WordPress/gutenberg/pull/18157#issuecomment-547973172).

Apart of that we do no deleted `icon` prop so you still can pass it (it is consistent with other BottomSheet Cells component) but now it shouldn't break layout. 

According to extra props please check the `BottomSheetCell`(file name is `cell.native.js`)  component. It is a base for other `Cells` in mobile version. It takes below props:
![image](https://user-images.githubusercontent.com/21242757/67923829-46f7bc00-fbaf-11e9-80eb-4b48c8e78f97.png)

However at this moment considering `RangeControl` component I think the only props that we might need to be able to expose are `accessibilityProps`. 

@pinarol 
Another idea is if we can add some logic to mentioned props and be able to automatically calculate them inside native component (e.g `separatorType` can be added automatically in `PanelBody` when rendering the children of BottomSheet Cells but then we loose a little bit control of manually changing it from `Inspector Control` block)

Currently the list of props to take care of looks as below
1. icon
2. separatorType
3. accessibilityLabel,
4. accessibilityHint,
5. accessibilityRole,

Reply:
That's the plan for separatorType, but we couldn't get to start working on [that](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1421) yet. And we don't need to do this as a part of this PR.

For these ones:

> accessibilityLabel,
> accessibilityHint,
> accessibilityRole,

I'd check if they crash web and make a fix accordingly but I wouldn't necessarily document them in README as they are available for all react-native components in general.

Reply:
@pinarol ok so I will:

1. Document mentioned props in README file
2. Make sure passing extra props specific to one platform do not break anything in other platform

Is there anything else that we should address in this PR ?

I also saw that Spacer feature was merged so I hope shortly we will also merge latest changes related to RangeCell made by Luke which was targeting Spacer branch

Reply:
> 1. Document mentioned props in README file

Yes, and it'd be enough if we only document `icon`, `separatorType`

> 2. Make sure passing extra props specific to one platform do not break anything in other platform

👍 

> Is there anything else that we should address in this PR ?

Not anything else that I can think of.

> I also saw that Spacer feature was merged so I hope shortly we will also merge latest changes related to RangeCell made by Luke which was targeting Spacer branch

👍

Reply:
I have update README (can you take a look if the description make sense ?), CI passed after sync latest `gutenberg/gutenberg-mobile` repo changes.

Now I'm testing if passing platform specific props do not break mobile/web app

Reply:
Thanks commented https://github.com/WordPress/gutenberg/pull/18036/files#r342016123